doc/README.md

```diff
@@ -114,3 +119,75 @@ pear
 watermelon
 :   green fruit with red flesh
 ```
+
+#### Examples
+
+To define a referenceable figure use the following fencing:
+
+```markdown
+:::{.example #an-attribute-set-example}
+# An attribute set example
+
+You can add text before
+
+    ```nix
+    { a = 1; b = 2;}
+    ```
+
+and after code fencing
+:::
+```
+
+Defining examples through the `example` fencing class adds them to a `List  of Examples` after the `Table of Contents`.
+
+#### Figures
+
+To define a referencable figure use the following fencing:
+
+```markdown
+::: {.figure #nixos-logo}
+# NixOS Logo
+![NixOS logo](./nixos_logo.png)
+:::
+```
+
+Defining figures through the `figure` fencing class adds them to a `List  of Figures` after the `Table of Contents`.
+
+#### Footnotes
+
+Footnotes are supported as a URL shortener alternative:
+
+```markdown
+::: {.note}
+Footnotes can be used to [shorten URLs][url-id] and keep the markdown readable.
+:::
+
+[url-id]: https://github.com/NixOS/nixpkgs/blob/19d4f7dc485f74109bd66ef74231285ff797a823/doc/README.md
+```
```

Comment:
I think that's just a [link reference definition](https://spec.commonmark.org/0.30/#link-reference-definitions), it doesn't appear in the rendered document